BibBuddy/app/src/main/java/de/bibbuddy/NoteRecyclerViewAdapter.java

```diff
@@ -220,6 +204,29 @@ private void setupMediaPlayerListeners(MediaPlayer mediaPlayer, SeekBarCompat pr
     });
   }
 
+  private void playOrPause(View v, MediaPlayer mediaPlayer, ImageButton stopButton,
+                       NoteItem noteItem, SeekBarListener seekBarListener){
+    ImageButton button = (ImageButton) v;
+
+    if (v.isSelected()) {
+      if (paused) {
+        mediaPlayerPosition = mediaPlayer.getCurrentPosition();
+        mediaPlayer.seekTo(mediaPlayerPosition);
+        mediaPlayer.start();
+        button.setImageResource(R.drawable.pause);
+      } else {
+        mediaPlayer.pause();
+        button.setImageResource(R.drawable.play);
+      }
+      paused = !paused;
+    } else {
+      resetPlayers();
+      startAudio(mediaPlayer, button, noteItem, seekBarListener);
+      stopButton.setClickable(true);
+    }
```

Comment:
I would add one or two new lines in this area to structure the code a bit :)

Reply:
You are right - this is done.